control/discovery.py

```diff
@@ -327,17 +327,36 @@ def __init__(self, config):
             assert 0
         self.logger.info(f"discovery addr: {self.discovery_addr} port: {self.discovery_port}")
 
+        self.sock = None
+        self.passed_enter = False
         self.conn_vals = {}
         self.connection_counter = 1
         self.selector = selectors.DefaultSelector()
 
     def __enter__(self):
+        self.passed_enter = True
```

Comment:
why do we need self.passed_enter? could we remove it?

Reply:
I added it to so I could assert in case someone started the discovery service not in a "with" block. As we now rely on the fact that this is how discovery for called. Otherwise we won't pass through __ exit__()